samtranslator/open_api/open_api.py

```diff
@@ -101,7 +101,14 @@ def get_integration_function_logical_id(self, path_name, method_name):
 
         # Extract lambda integration (${LambdaName.Arn}) and split ".Arn" off from it
         regex = "([A-Za-z0-9]+\.Arn)"
-        match = re.findall(regex, arn)[0].split(".Arn")[0]
+        matches = re.findall(regex, arn)
+        # For Functions with AutoPublishAlias the integration URI will contain
+        # ${LambdaName}, not ${LambdaName.Arn} (both resolve to the AWS::Lambda::Alias ARN).
+        # Without the .Arn suffix, parsing the URI is more ambiguous, so instead of
+        # introducing more brittle hacks, return gracefully.
+        if not matches:
+            return False
```

Comment:
Does this mean the integration is not setup then? Can we add and End to End tests to verify this actually fixes the issues and produces a deploy-able template?

Reply:
Added